src/distrs/wilcox.jl

```diff
@@ -0,0 +1,149 @@
+#=
+The wilcox distribution is equivalent to the problem
+of finding the number of subsets with size nx
+of {1,2,...,nx,nx+1,...,nx+ny} summing to (U+sum(1:nx))
+relative to the total number of subsets with size nx.
+The empty subset is defined to sum to zero.
+This can be calculated using the recursion:
+either nx+ny is in the subset in which case we need to calculate
+the number of subsets with size nx-1 of {1,2,...,nx,nx+1,...,nx+ny-1}
+summing to (U+sum(1:nx)-nx-ny),
+or nx is not in the subset in which case we need to calculate
+the number of subsets with size nx of {1,2,...,nx,nx+1,...,nx+ny-1}
+summing to (U+sum(1:nx)),
+This can be calculated bottom up using dynamic programming.
+
+The (i,j)'th element of DP in the k'th outer loop iteration represents:
+the number of subsets with size k of {1,2,...,k,k+1,...,k+j-1} summing to i+sum(1:k)-1.
+ =#
+
+@inline function wilcoxDP(nx, ny, U)
+    DP = zeros(Int, U + 1, ny + 1)
+    for j = 1:(ny+1)
+        DP[1, j] = 1
+    end
+    for k = 1:nx
+        for j = 2:(ny+1)
+            for i = (U+1):-1:2
+                DP[i, j] = DP[i, j-1]
+                if i - j >= 0
+                    DP[i, j] += DP[i-j+1, j]
+                end
+            end
```

Comment:
I'd use the version without `if`, also to reduce memory accesses to `DP[i, j]`:
```suggestion
            for i = (U+1):-1:j
                DP[i, j] = DP[i-j+1, j] + DP[i, j-1] 
            end
            for i = (j-1):-1:2
                DP[i, j] = DP[i, j-1] 
            end
```

Reply:
`(U+1)` is a loose upper bound for the iteration of `i`, do you know the correct tight bound?
I think it is something like:
`for i = min((U+1),sum((j):(k+j-1))-sum(1:k)+1):-1:j`

Reply:
The splitting of the loop doesn't work, because the start of the second loop can be before the start of the first, i.e. it is not guaranteed that `j` lies between `(U+1)` and `2`.
e.g. if DP is a 1 by 5 matrix, then the second loop will go out of bounds.

Reply:
> do you know the correct tight bound?

I haven't thought more about it yet 😄 

> The splitting of the loop doesn't work, because the start of the second loop can be before the start of the first, i.e. it is not guaranteed that j lies between (U+1) and 2.

Ah, my bad, I think it should have been
```suggestion
            for i = (U+1):-1:max(j,2)
                # In this loop: U+1 >= i >= 2 AND i - j >= 0
                DP[i, j] = DP[i-j+1, j] + DP[i, j-1] 
            end
            for i = min(U+1,j-1):-1:2
                # In this loop: U+1 >= i >= 2 AND i - j < 0
                DP[i, j] = DP[i, j-1] 
            end
```

Reply:
> do you know the correct tight bound?

As you wrote in the comment in the code,

> The (i,j)'th element of DP in the k'th outer loop iteration represents:
the number of subsets with size k of {1,2,...,k,k+1,...,k+j-1} summing to i+sum(1:k)-1.

Let q be the maximum sum of a subset of size k of {1,2,...,k+j-1} that may be attained. We know about q:
1. q is less than or equal to the sum of the subset {j,j+1,...,k+j-1} (k largest values in the set), ie. q <= k*(2*j+k-1)/2.
2. Any subset of size nx that consists of the subset that sums up to q and a subset of size nx-k of {k+j,k+j+1,...,nx+ny} sums up to at least q+sum((k+j):(nx+j-1)) (nx-k smallest numbers in {k+j,k+j+1,...}). Since we only consider sets of size nx that sum up to U+sum(1:nx), this implies that q+sum((k+j):(nx+j-1)) <= U+sum(1:nx), or equivalently q <= U+k*(2*j+k-1)/2-nx*j.

Taken, together we have q <= min(k*(2*j+k-1)/2, U+k*(2*j+k-1)/2-nx*j) = k*(2*j+k-1)/2+min(U-nx*j,0).

Thus i+sum(1:k)-1 <= q <= k*(2*j+k-1)/2+min(U-nx*j,0), and hence i <= (j-1)*k+1+min(U-nx*j,0).

Reply:
Ok, I think I included all this in the latest commit.
Please do a final review of the tricky indexing here, and then hopefully this is good to go.